exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/util/DsosMapperUtil.java

```diff
@@ -2,39 +2,73 @@
 
 import java.util.Arrays;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntFunction;
 import java.util.function.Predicate;
 
 public class DsosMapperUtil {
     public static final int DEFAULT_DAYS_SINCE_SYMPTOMS = 4000;
+    public static final int DEFAULT_LOCAL_DAYS_SINCE_SYMPTOMS = 0;
     private static final Integer DEFAULT_VALUE = null;
 
     public enum DsosInterpretationMapper {
-        SYMPTOMS_WITH_DATE(dsos -> -14 <= dsos && dsos <= 14, dsos -> dsos),
-        SYMPTOMS_WITH_TIME_RANGE(dsos -> 100 <= dsos && dsos <= 1900, dsos -> DEFAULT_VALUE),
-        SYMPTOMS_WITHOUT_DATE(dsos -> 1986 <= dsos && dsos <= 2014, dsos -> DEFAULT_VALUE),
-        SYMPTOMS_UNKNOWN(dsos -> 3986 <= dsos && dsos <= 4014, dsos -> DEFAULT_VALUE),
-        NO_SYMPTOMS(dsos -> 2986 <= dsos && dsos <= 3014, dsos -> DEFAULT_VALUE),
-        UNKNOWN(dsos -> Integer.MAX_VALUE == dsos, dsos -> DEFAULT_VALUE);
+        SYMPTOMS_WITH_DATE(dsos -> -14 <= dsos && dsos <= 14, dsos -> dsos, 0),
+        SYMPTOMS_WITH_TIME_RANGE(dsos -> 100 <= dsos && dsos <= 1900, dsos -> DEFAULT_VALUE, 1400),
+        SYMPTOMS_WITHOUT_DATE(dsos -> 1986 <= dsos && dsos <= 2014, dsos -> DEFAULT_VALUE, 2000),
+        SYMPTOMS_UNKNOWN(dsos -> 3986 <= dsos && dsos <= 4014, dsos -> DEFAULT_VALUE, 4000),
+        NO_SYMPTOMS(dsos -> 2986 <= dsos && dsos <= 3014, dsos -> DEFAULT_VALUE, 3000),
+        UNKNOWN(dsos -> Integer.MAX_VALUE == dsos, dsos -> DEFAULT_VALUE, DEFAULT_LOCAL_DAYS_SINCE_SYMPTOMS);
 
         private final Predicate<Integer> range;
         private final IntFunction<Integer> mapper;
+        private final int zero;
 
-        DsosInterpretationMapper(Predicate<Integer> range, IntFunction<Integer> mapper) {
+        DsosInterpretationMapper(Predicate<Integer> range, IntFunction<Integer> mapper, int zero) {
```

Comment:
zero?

Reply:
yep, used just zero, because wasn't sure what is proper term in English. But means "zero" point in current mapping, e.g. in SYMPTOMS_UNKNOWN zero point is 4000.

Reply:
An yep, there's not actual zero point in all of the mappings, but those are not used currently, so it's not the cleanest, but..